### ISSUE TITLE
Rule S2223 FP when using ThreadStatic

### ISSUE THREAD

**Issue by costin-zaharia-sonarsource** [SonarSource/sonar-dotnet#4407]
### Description

Rule [S2223](https://rules.sonarsource.com/csharp/RSPEC-2223) is raising a false positive when using ThreadStatic attributes on fields.

### Repro steps

```
using System;

class Program
{
    [ThreadStatic]
    public static int value;

    static void Main(){}
}
```

### Expected behavior

S2223 should not be raised in this case since the field is decorated with ThreadStatic attribute and it's value is not shared between threads.

### Actual behavior

S2223 should is raised.

See: https://discuss.sonarsource.com/t/s2223-and-threadstatic-attribute/7517/3
